### PR TITLE
Use flag to control use_horovod.

### PR DESCRIPTION
**Pull request by mingxingtan**
Followup to https://github.com/google/automl/pull/413

### CODE REVIEW

File: efficientdet/main.py [hunk 1]
```diff
@@ -131,6 +121,15 @@
 
 
 def main(_):
+
+  if FLAGS.use_horovod:
```
Comment (mingxingtan) [inline efficientdet/main.py:125]: @fsx950223 use a flag to control whether to use horovod.

**Comment by fsx950223** [comment]
According to paper, batch norm should replace by sync batch norm.
```python
class SyncBatchNormalization(tf.keras.layers.BatchNormalization):
  """Cross replica batch normalization."""

  def __init__(self, fused=False, **kwargs):
    if fused in (True, None):
      raise ValueError('SyncBatchNormalization does not support fused=True.')
    if not kwargs.get('name', None):
      kwargs['name'] = 'tpu_batch_normalization'
    super(SyncBatchNormalization, self).__init__(fused=fused, **kwargs)

  def _moments(self, inputs, reduction_axes, keep_dims):
    """Compute the mean and variance: it overrides the original _moments."""
    import horovod as hvd
    shard_mean, shard_variance = super(SyncBatchNormalization, self)._moments(
      inputs, reduction_axes, keep_dims=keep_dims)

    num_shards = hvd.size()
    if num_shards > 1:
      # Compute variance using: Var[X]= E[X^2] - E[X]^2.
      shard_square_of_mean = tf.math.square(shard_mean)
      shard_mean_of_square = shard_variance + shard_square_of_mean
      group_mean = hvd.allreduce(shard_mean)
      group_mean_of_square = hvd.allreduce(shard_mean_of_square)
      group_variance = group_mean_of_square - tf.math.square(group_mean)
      return (group_mean, group_variance)
    else:
      return (shard_mean, shard_variance)

  def call(self, *args, **kwargs):
    outputs = super(SyncBatchNormalization, self).call(*args, **kwargs)
    # A temporary hack for tf1 compatibility with keras batch norm.
    for u in self.updates:
      tf.add_to_collection(tf.GraphKeys.UPDATE_OPS, u)
    return outputs
```
It needs to add ```use_horovod``` to every batch norm function. Any better solution?

**Comment by mingxingtan** [comment]
Good point. Maybe you can pass use_horovod around (similar to use_tpu)?  (Or you can also unify them to be a common flag and pass around the common flag.

Does tf.keras.layers.experimental.SyncBatchNormalization supporrt horovod?

**Comment by fsx950223** [comment]
> Good point. Maybe you can pass use_horovod around (similar to use_tpu)? (Or you can also unify them to be a common flag and pass around the common flag.
> 
> Does tf.keras.layers.experimental.SyncBatchNormalization supporrt horovod?

```tf.keras.layers.experimental.SyncBatchNormalization``` doesn't horovod and it's a experimental layer only supports latest TensorFlow.

**Comment by mingxingtan** [comment]
@fsx950223 Your horovod sync batch norm looks good to me. Do you plan to send a PR, so I can merge this to master branch?

**Comment by fsx950223** [comment]
> @fsx950223 Your horovod sync batch norm looks good to me. Do you plan to send a PR, so I can merge this to master branch?

Yes, I do it later.
